### ISSUE TITLE
No ability to set triplet dynamic/static flavors

### ISSUE THREAD

**Issue by anthonyalayo** [microsoft/vcpkg#35994]
**Describe the bug**
I started a discussion on this to confirm the ability doesn't exist:
https://github.com/microsoft/vcpkg/discussions/35987#discussioncomment-7996426

CMake is the standard build generator, and the integration with CMake is great. However, the integration currently doesnt allow users to specify static vs dynamic libraries _dynamically_. Modern CMake uses `BUILD_SHARED_LIBS` as a global flag. Ideally there would be a straightforward way to use this to select between dynamic and static triplets automatically.

In particular, I am manging a build with OSX and Linux. I have support for static and dynamic builds, yet there is no way for Vcpkg to simply adjust the library build type.

**Expected behavior**
There would be some sort of `VCPKG_TRIPLET_SUFFIX` that I can set to `dynamic` if `BUILD_SHARED_LIBS` is set.

**Comment by Osyotr** [comment]
> `BUILD_SHARED_LIBS`

This option affects only **your** targets. It doesn't and shouldn't (and can't reliably) control external dependencies.

**Comment by anthonyalayo** [comment]
> > `BUILD_SHARED_LIBS`
> 
> This option affects only **your** targets. It doesn't and shouldn't (and can't reliably) control external dependencies.

@Osyotr definitely, I agree, but there should be some way to change the triplet to dynamic if `BUILD_SHARED_LIBS` is set. Otherwise static libs get built by default, and the mix of shared and static can cause symbol resolution errors.

**Comment by anthonyalayo** [comment]
Adding more color, at present I would need to do something like copy/paste all of this:
https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/vcpkg.cmake#L233-L376

Just to then add some check on `BUILD_SHARED_LIBS` afterwards.

**Comment by Osyotr** [comment]
Please don't try to outsmart your users and fellow colleagues. Just create a cmake preset that sets appropriate `VCPKG_TARGET_TRIPLET` for configurations you care about.

**Comment by anthonyalayo** [comment]
> Please don't try to outsmart your users and fellow colleagues.

I am not following. How so? I am simply asking for a feature that makes sense, as I looked for it and couldn't find it.

> Just create a cmake preset that sets appropriate VCPKG_TARGET_TRIPLET for configurations you care about.

I just commented on this idea in the linked discussion. AFAIK there is no way to automatically apply presets. So a user would have to manually set presets based on their architecture / OS / platform. This doesn't make sense. We already have logic to determine this in CMake itself.
